scipy/stats/tests/test_distributions.py

```diff
@@ -1862,6 +1862,50 @@ def test_logcdf_logsf(self):
         logsf = stats.invgauss.logsf(110, 1.05)
         assert_allclose(logsf, -56.1467092416426)
 
+    def test_pdf(self):
+        # regression tests for improvements introduces in gh-13665
+        # these tests are bases on Gyner & Smyth (2016)
+
+        # infinite mean corresponds to an inverse-chisquare pdf.
+        assert_allclose(
+            stats.invgauss.pdf([-1, 0, 1, 2, np.inf, np.nan], mu=np.inf),
+            [0., 0., 0.24197072, 0.10984782, 0., np.nan]
+        )
+        # extreme values have zero density for any finite mean value
+        assert_allclose(stats.invgauss.pdf([-1, 0, np.inf], mu=1), [0, 0, 0])
+
+    def test_ppf_sf(self):
+        p = [0.000001, 0.00001, 0.0001, 0.001, 0.01, 0.1, 0.5, 0.9, 0.99,
+             0.999, 0.9999, 0.99999, 0.999999]
+        p1 = stats.invgauss.cdf(stats.invgauss.ppf(p, mu=1), mu=1)
+        abs_error = np.abs(p - p1)
+        # five-number summary of the absolute error
+        fns = np.percentile(abs_error, [0, 25, 50, np.mean(abs_error), 75, 100])
+        assert_allclose(fns, [0., 0., 8.13151629e-20, 0.,
+                        5.55111512e-17, 2.22044605e-16])
```

Comment:
Yes, you can set the `atol` of [`assert_allclose`](https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_allclose.html).

Reply:
What's the recommended value? should I make it tighter? 1e-12?

Reply:
Pytest is reporting a maximum absolute difference of `1.11022303e-16`, and that appears to be on a very tiny value, so absolute tolerance (rather than relative tolerance) is what we need to adjust. The default `atol` is 0, so that needs to be raised to at least as big as the maximum absolute difference - preferably a little bigger in case CI machines change a little bit. `atol=1e-15` would probably work, but if you wouldn't be concerned with an error `1e-12`, sure, you could change it to that.

Reply:
it looks like it worked, thanks. But now im getting another one:
```
=================================== FAILURES ===================================
___________________________ TestInvgauss.test_ppf_sf ___________________________
[gw0] linux -- Python 3.7.10 /opt/hostedtoolcache/Python/3.7.10/x64/bin/python
scipy/stats/tests/test_distributions.py:1903: in test_ppf_sf
    assert_allclose(stats.invgauss.ppf([0, 0.5, 1, 2, np.nan], mu=1),
        abs_error  = array([4.23516474e-22, 2.03287907e-20, 8.13151629e-20, 2.16840434e-19
       12.0318933 , 15.90115227, 19.90009758])
        q1         = array([ 0.03872821,  0.04676404,  0.0588942 ,  0.07921848,  0.11984124,
        0.23762471,  0.67584131,  2.14303391,  4.98409484,  8.35486493,
       12.0318933 , 15.90115227, 19.90009758])
        rel_error  = array([0.00000000e+00, 1.48380963e-16, 1.17819649e-16, 1.75183722e-16,
       0.00000000e+00, 3.50412747e-16, 0.00000000e+00, 8.28898147e-16,
       1.24741786e-15, 1.80721451e-14, 0.00000000e+00, 0.00000000e+00,
       1.68867330e-11])
        self       = <scipy.stats.tests.test_distributions.TestInvgauss object at 0x7f62f8725290>
scipy/stats/_distn_infrastructure.py:2126: in ppf
    cond1 = (0 < q) & (q < 1)
E   RuntimeWarning: invalid value encountered in greater
        _a         = 0.0
        _b         = inf
        args       = (array(1),)
        cond0      = True
        kwds       = {'mu': 1}
        loc        = array(0)
        q          = array([0. , 0.5, 1. , 2. , nan])
        scale      = array(1)
        self       = <scipy.stats._continuous_distns.invgauss_gen object at 0x7f634cd9b710>

```

It looks like assert_allclose() does not like comparisons with nan in them?

Reply:
I don't think it's assert_allclose that is complaining. It seems like it's something in the PPF function. `assert_allclose` can compare NaN to NaN:
```python3
import numpy as np
np.testing.assert_allclose(np.nan, np.nan)
np.testing.assert_allclose(np.nan, 1)
```

Reply:
Are these failures platform dependent? It seems that the failures occur on windows builds and linux 32bit.